include/dnscpp/context.h

```diff
@@ -46,17 +46,21 @@ class Context : private Core
      *  you can run any queries.
      *  @param  loop        your event loop
      *  @param  defaults    should system settings be loaded
+     *  @param  buffersize  size of the send & receive buffers of UDP sockets. Use 0 for system default
+     *  @param  socketcount number of UDP sockets to maintain
      */
-    Context(Loop *loop, bool defaults = true) : Core(loop, defaults) {}
+    Context(Loop *loop, bool defaults = true, int32_t buffersize = 0, size_t socketcount = 1) : Core(loop, defaults, buffersize, socketcount) {}
```

Comment:
Do not create a constructor with an endless number of options. The options can be supplied via the methods.

Reply:
> The options can be supplied via the methods.

This would require resizing a possibly active sockets array. Can you explain how that's supposed to work? We would have to cancel all lookups.